src/components/header/Header.tsx

```diff
@@ -42,13 +44,22 @@ const StyledHeader = styled.div`
 const StyledIconDiv = styled.div`
   display: flex;
   align-items: center;
+  .badge-icon {
+    top: 34px;
+    right: 40px;
+  }
 `;
 
 export const Header: React.FC = () => {
   // const { user } = useAuth0();
-
+  const { cart } = useContext(CartContext);
+  const [totalQuantity, setTotalQuantity] = useState(0);
   const [searchIsActive, setSearchIsActive] = useState<boolean>(false);
-
+  useEffect(() => {
+    let total = 0;
+    cart.map((items) => (total += items.quantity));
+    setTotalQuantity(total);
+  });
```

Comment:
Woahkay we can simplify this a bit. Do you know about [useMemo](https://reactjs.org/docs/hooks-reference.html#usememo)? It's a hook for doing expensive calculations (like array manipulation) only when their dependencies change rather than every render. Rather than having another state and an effect you can just do 
```
...
  const { cart } = useContext(CartContext);
  const [searchIsActive, setSearchIsActive] = useState<boolean>(false);
  const cartItemsCount = useMemo(() => {
    return cart.reduce((count, { quantity }) => count + quantity, 0);
  }, [cart]);
  return (
...
```